### PR TITLE
UI rework

### PR DESCRIPTION
**Pull request by georgebuciuman**
### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Added `subtitle` property to various components and interfaces to enhance UI.
- Updated styles in multiple components for better UI consistency.
- Fixed the selector used for downloading the graph.
- Added `mainGraphSelector` to select the main graph state.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>handles.tsx</strong><dd><code>Add minWidth property to Stack component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-editor/src/components/flow/handles.tsx
<li>Added <code>minWidth</code> property to <code>Stack</code> component for better UI consistency.<br>

</details>
    

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/278/files#diff-8894a9f9739633943a3bb8589e433b347119e6d70213e1104139d05c634b1b53">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>node.tsx</strong><dd><code>Add subtitle property and update Node styles</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-editor/src/components/flow/wrapper/node.tsx
<li>Added <code>subtitle</code> property to <code>NodeProps</code> interface.<br> <li> Updated styles for <code>NodeWrapper</code> and <code>Node</code> components.<br> <li> Included <code>subtitle</code> rendering in <code>Node</code> component.<br>

</details>
    

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/278/files#diff-9f5931cfdd7cdc0c4c8706ac2c131b74522f9a22bac3026a903387ca828815e3">+26/-18</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>nodeV2.tsx</strong><dd><code>Add subtitle property to NodeV2 components</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-editor/src/components/flow/wrapper/nodeV2.tsx
<li>Added <code>subtitle</code> property to <code>UiNodeDefinition</code> and <code>INodeWrap</code> interfaces.<br> <li> Updated <code>NodeWrap</code> component to handle <code>subtitle</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/278/files#diff-22c64f79290285f9b5524bf34746fe5aba1d1ef1100277adfd1fd2403621c07d">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>graph.ts</strong><dd><code>Add mainGraphSelector to graph selectors</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-editor/src/redux/selectors/graph.ts
- Added `mainGraphSelector` to select the main graph state.

</details>
    

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/278/files#diff-551fd55f25a1503d4c4117d884c5605f8fbedf25f79f07f2800a25912b07b146">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>reactflow.css</strong><dd><code>Update CSS styles for React Flow components</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-editor/src/css/reactflow.css
- Updated various CSS properties for better UI consistency.

</details>
    

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/278/files#diff-97510174aa013a0aa559154d3b0df384d8403f5c9cdf5f4b31b1d3082acfbf13">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>defaults.tsx</strong><dd><code>Fix graph download selector</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-editor/src/components/menubar/defaults.tsx
- Fixed the selector used for downloading the graph.

</details>
    

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/278/files#diff-73e44b3fee24b066b3bb9345ff46225f892d46fd67f3d91be8306dc054e954ac">+3/-5</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/graph-editor/src/components/flow/handles.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const HandleContainer = ({
       <Stack
         direction="column"
         gap={2}
-        css={{ flexBasis: full ? '100%' : '50%', position: 'relative', textAlign: type === 'source' ? 'right' : 'left'}}
+        css={{ flexBasis: full ? '100%' : '50%', position: 'relative', textAlign: type === 'source' ? 'right' : 'left', minWidth: '250px' }}
```
Comment (SorsOps) [inline packages/graph-editor/src/components/flow/handles.tsx:47]: I'm assuming this is to account for the inline preview?

File: packages/graph-editor/src/redux/selectors/graph.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ export const logSelector = createSelector(graph, (state) => state.logs);
 
 export const playStateSelector = createSelector(graph, (state) => state.graphPlayState);
 
+export const mainGraphSelector = createSelector(graph, (state) => state.panels?.graph1);
```
Comment (SorsOps) [inline packages/graph-editor/src/redux/selectors/graph.ts:12]: Moving forward this feels too brittle

File: packages/graph-editor/src/components/flow/handles.tsx (hunk 1 above)
Comment (georgebuciuman) [reply packages/graph-editor/src/components/flow/handles.tsx:47]: `textAlign` is for inline preview. I don't know exactly what `minWidth` is for. @mck could you help us with the reason for this?

File: packages/graph-editor/src/redux/selectors/graph.ts (hunk 2 above)
Comment (georgebuciuman) [reply packages/graph-editor/src/redux/selectors/graph.ts]: We were hardcoding graph id everywhere. I added a constant and used it

**Review by roppazvan — APPROVED** [review]

**Pull request merged**